BlackMarket/app/Components/BMTextFieldView.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  BMTextFieldView.swift
+//  BlackMarket
+//
+//  Created by Tarek Radovan on 06/12/2022.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct BMTextFieldDomain: ReducerProtocol {
+  struct State: Equatable {
+    @BindableState var text: String = ""
+    var validations = [ValidationType]()
+    var placeholder: String?
+    var errorMessage: String?
+    var title: String = ""
+    var isSecure: Bool = false
+    var isValid: Bool {
+      text.validates(validations)
+    }
+    var shouldShowError: Bool {
+      !isValid && !text.isEmpty
```

Comment:
We already have a `ValidationType.nonEmpty`, so this should not be explicitly validated if the validation is not present in the array right?